### PR TITLE
remove legacy `selectAssets` functionality

### PR DESCRIPTION
**Pull request by Unisay**
This PR removes legacy coin selection functionality that isn't used anymore. 

It also rewrites benchmarks to use new `transactionFee` functionality for benchmarking instead of the old unused `selectAssets`.

### Issue Number

ADP-2678

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet.hs [hunk 1]
```diff
@@ -1793,141 +1772,6 @@ calcMinimumCoinValues ctx era outs = do
...
--- randomly.
---
--- By default, the seed used for random selection is derived automatically,
--- from the given 'MonadRandom' context.
---
--- However, if a concrete value is specified for the optional 'randomSeed'
--- parameter, then that value will be used instead as the seed for random
--- selection.
---
-selectAssets
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Wallet.hs:1830]: 👏

File: lib/wallet/bench/restore-bench.hs [hunk 2]
```diff
@@ -587,29 +586,22 @@ benchmarksSeq network w wid _wname benchname restoreTime = do
...
-                , wallet
-                , selectionStrategy = SelectionStrategyOptimal
-                } $ \_state -> W.Fee . selectionDelta TokenBundle.getCoin
-        runExceptT $ withExceptT show $ W.calculateFeePercentiles estimateFee
+        AnyRecentEra (recentEra :: WriteTx.RecentEra era) <-
+            guardIsRecentEra =<< currentNodeEra netLayer
+        (protocolParameters, _bundledProtocolParameters) <-
+            W.toBalanceTxPParams @era <$> currentProtocolParameters netLayer
+        timeTranslation <- toTimeTranslation (timeInterpreter netLayer)
+        W.transactionFee @s @k @n
```
Comment (Anviking) [inline lib/wallet/bench/restore-bench.hs:594]: 👌

**Review by Anviking — APPROVED** [review]
lgtm

**Comment by Unisay** [comment]
bors merge

**Pull request merged**